src/filters.cpp

```diff
@@ -157,9 +157,36 @@ NAN_METHOD(Filters::New) {
                     // insert the key/value into filters map
                     // TODO(dane): what if we have duplicate source-layer filters?
 
+                    // handle property array
+                    const v8::Local<v8::Value> layer_properties = layer->Get(Nan::New("properties").ToLocalChecked());
+                    if (layer_properties->IsNull() || layer_properties->IsUndefined()) {
+                        Nan::ThrowError("Property-Filters is not properly constructed.");
+                        return;
+                    }
+
+                    // NOTICE: If a layer is styled, but does not have a property, the property value will equal []
+                    // NOTICE: If a property is true, that means we need to keep all the properties
+                    filter_properties property;
+                    if (layer_properties->IsArray()) {
```

Comment:
yeah, In short words: The `filters` filert in styleTofilters.js use `true` for keep all. I've also used true to represent everything in order to be consistent.

------

more details: 
`filter_properties` means which property **to keep**,not which property **to remove**, so follow this idea, the empty means keep nothing, if we make it means keep all properties, that's a bit of a contradiction.

So, while I design this, I thought: In default, we don't need to keep any properties, but we need to collection the properties names since we start the traverse,so I use an array to store the properties which we need to keep. Only when we meat the key words `"properties"` which means we need to keep all the properties.  Follow the `filters` reules, I use `ture` to make this. 

so, true means every properties, `filter_properties` means which to keep, empty array means nothing to keep.

Reply:
👍 Sounds good. Thanks for explaining.